### PR TITLE
ddl: add lease for ingest distributed lock

### LINKED ISSUE #55917: add index blocked after injection pd leader io delay
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、tidb_enable_dist_task='off'
2、run sysbench
3、add index for one table
4、injection pd leader io delay

logs:
[endless-ha-test-add-index-tps-7553489-1-251.zip](https://github.com/user-attachments/files/16904374/endless-ha-test-add-index-tps-7553489-1-251.zip)

### 2. What did you expect to see? (Required)
add index can success

### 3. What did you see instead (Required)
add index blocked after injection pd leader io delay

img_v3_02ef_94d2d15f-1519-4f64-9b90-07b74210aebg

the status of ddl job is not synced after 1h0m0s (now: 2024-09-06 12:33:26, jobId: 611, job type: add index /* ingest */, state: running)
operatorLogs:
[2024-09-06 11:33:15] ###### start adding index
ALTER TABLE `sbtest1` ADD INDEX `index_test_1725593595755`(`c`)
[2024-09-06 11:33:15] ###### wait for ddl job finish
[2024-09-06 12:33:26] ###### wait for ddl job finish timeout(1h0m0s)
select job_id, db_name, table_name, job_type, create_time, start_time, end_time, state, query from information_schema.ddl_jobs where query = 'ALTER TABLE `sbtest1` ADD INDEX `index_test_1725593595755`(`c`)'
jobId: 611, job type: add index /* ingest */, state: running

### 4. What is your TiDB version? (Required)
./tidb-server -V
 Release Version: v8.4.0-alpha
Edition: Community
Git Commit Hash: 9c7f4eb455a4e781e05da4f4125c49fda29a74f5
Git Branch: heads/refs/tags/v8.4.0-alpha
UTC Build Time: 2024-09-05 11:51:24
GoVersion: go1.21.10
Race Enabled: false
Check Table Before Drop: false
Store: unistore
2024-09-06T11:13:31.856+0800

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #55917

Problem Summary:

The distributed lock was acquired successfully, but it failed to be released. When DDL retried in the next round, it stuck at acquiring the same distributed lock.

### What changed and how does it work?

Similar to existing mechanisms of DDL owner key or server info key, add a lease for the distributed lock

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

**Review by lance6716 — COMMENTED** [review]
add some unit tests using embedded etcd?

https://github.com/search?q=repo%3Apingcap%2Ftiflow%20embed.StartEtcd&type=code

File: pkg/owner/manager.go [hunk 1]
```diff
@@ -499,3 +499,31 @@ func init() {
...
 }
+
+// AcquireDistributedLock creates a mutex with ETCD client, and returns a mutex release function.
+func AcquireDistributedLock(
+	ctx context.Context,
+	cli *clientv3.Client,
+	key string,
+	ttlInSec int,
+) (release func(), err error) {
+	se, _ := concurrency.NewSession(cli, concurrency.WithTTL(ttlInSec))
```
Comment (lance6716) [inline pkg/owner/manager.go:510]: check the error?

File: pkg/owner/manager.go [hunk 2]
```diff
@@ -499,3 +499,31 @@ func init() {
...
+	mu := concurrency.NewMutex(se, key)
+	err = mu.Lock(ctx)
+	if err != nil {
```
Comment (lance6716) [inline pkg/owner/manager.go:513]: the session is not closed

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: pkg/owner/manager.go (hunk 1 above)
Comment (tangenta) [reply pkg/owner/manager.go]: https://github.com/pingcap/tidb/pull/56184/commits/5f4d438b232019de96a10495d721b8fd90d3eb5c

File: pkg/owner/manager.go (hunk 2 above)
Comment (tangenta) [reply pkg/owner/manager.go:516]: https://github.com/pingcap/tidb/pull/56184/commits/5f4d438b232019de96a10495d721b8fd90d3eb5c

**Review by lance6716 — APPROVED** [review]

**Pull request merged**